azure/services/fleetsmember/spec.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fleetsmember
+
+import (
+	"context"
+
+	asocontainerservicev1 "github.com/Azure/azure-service-operator/v2/api/containerservice/v1api20230315preview"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/cluster-api-provider-azure/azure"
+)
+
+// AzureFleetsMemberSpec defines the specification for azure fleet feature.
+type AzureFleetsMemberSpec struct {
+	Name                 string
+	Namespace            string
+	ResourceGroup        string
```

Comment:
since this is a child resource to Fleet, it cannot be in a different resourcegroup than the Fleet.
is this necessary?

the ClusterResourceReference is the part that provides teh RG of the target AKS cluster. so this seems redundant with "ManagerResourceGroup"?

Reply:
`ManagerResourceGroup` is the resource group for the Azure Fleet Manager, not the cluster. We still need the resource group for the current cluster to use in the `ClusterResourceReference`.

Reply:
Ah I think that would be cleaner, will rename the field. As for the naming, is there any reason to name the member something different from the cluster? If it is useful to the user to name it something different than the cluster name, I think we can go ahead and add that field.

Reply:
As I said in a different comment, 2 cluster can have the same name if they are in a different subscription or in a different resourcegroup.

I can have multiple RG:
- Europe
- US
- Asia

And in each have some clusters named the same:

- cluster-staging
- cluster-prod

For example.

The clusters all have the same name, but you can't have them all with member name = cluster name.

You can do something smart, but you won't escape letting the user provide a different name.

The name of the member is also available for workload placement, as a constraint. And the user might want to name it differently.

Reply:
That makes sense. I added a new field for the name of the member which defaults to the cluster name if not specified: https://github.com/willie-yao/cluster-api-provider-azure/blob/aks-fleets-member/api/v1beta1/types.go#L1012-L1013